op-e2e/actions/ecotone_fork_test.go

```diff
@@ -0,0 +1,139 @@
+	// See [derive.EcotoneNetworkUpgradeTransactions]
+	require.Equal(t, 7, len(transactions))
+
+	// All transactions are successful
+	for i := 1; i < 7; i++ {
+		txn := transactions[i]
+		receipt, err := engine.EthClient().TransactionReceipt(context.Background(), txn.Hash())
+		require.NoError(t, err)
+		require.Equal(t, types.ReceiptStatusSuccessful, receipt.Status)
+	}
+
+	expectedL1BlockAddress := crypto.CreateAddress(derive.L1BlockDeployerAddress, 0)
+	expectedGasPriceOracleAddress := crypto.CreateAddress(derive.GasPriceOracleDeployerAddress, 0)
+
+	// Gas Price Oracle Proxy is updated
+	updatedGasPriceOracleAddress, err := engine.EthClient().StorageAt(context.Background(), predeploys.GasPriceOracleAddr, genesis.ImplementationSlot, latestBlock.Number())
+	require.NoError(t, err)
+	assert.Equal(t, expectedGasPriceOracleAddress, common.BytesToAddress(updatedGasPriceOracleAddress))
+	assert.NotEqualf(t, initialGasPriceOracleAddress, updatedGasPriceOracleAddress, "Gas Price Oracle Proxy address should have changed")
+
+	// L1Block Proxy is updated
+	updatedL1BlockAddress, err := engine.EthClient().StorageAt(context.Background(), predeploys.L1BlockAddr, genesis.ImplementationSlot, latestBlock.Number())
+	require.NoError(t, err)
+	assert.Equal(t, expectedL1BlockAddress, common.BytesToAddress(updatedL1BlockAddress))
+	assert.NotEqualf(t, initialL1BlockAddress, updatedL1BlockAddress, "L1Block Proxy address should have changed")
+
+	_, err = gasPriceOracle.Scalar(nil)
+	require.ErrorContains(t, err, "scalar() is deprecated")
+
+	cost, err := gasPriceOracle.GetL1Fee(nil, []byte{0, 1, 2, 3, 4})
+	require.NoError(t, err)
+	require.Equal(t, cost.Uint64(), uint64(0), "expecting zero scalars within activation block")
+
+	// 4788 contract is deployed
+	expected4788Address := crypto.CreateAddress(derive.EIP4788From, 0)
+	code, err := engine.EthClient().CodeAt(context.Background(), expected4788Address, latestBlock.Number())
+	require.NoError(t, err)
+	require.Equal(t, code, predeploys.EIP4788ContractCode)
+
+	// Build empty L2 block, to pass ecotone activation
+	sequencer.ActL2StartBlock(t)
+	sequencer.ActL2EndBlock(t)
+
+	// assert again, now that we are past activation
+	_, err = gasPriceOracle.Scalar(nil)
+	require.ErrorContains(t, err, "scalar() is deprecated")
+
+	// test if the migrated scalar matches the deploy config
+	basefeeScalar, err := gasPriceOracle.BaseFeeScalar(nil)
+	require.NoError(t, err)
+	require.True(t, uint64(basefeeScalar) == dp.DeployConfig.GasPriceOracleScalar, "must match deploy config")
+
+	cost, err = gasPriceOracle.GetL1Fee(nil, []byte{0, 1, 2, 3, 4})
+	require.NoError(t, err)
+	require.Greater(t, cost.Uint64(), uint64(0), "expecting non-zero scalars after activation block")
+
+	// Get L1Block info
+	l1Block, err := bindings.NewL1BlockCaller(predeploys.L1BlockAddr, engine.EthClient())
+	require.NoError(t, err)
+	l1BlockInfo, err := l1Block.Timestamp(nil)
+	require.NoError(t, err)
+	assert.Greater(t, l1BlockInfo, uint64(0))
```

Comment:
similar to above only would you mind also adding:

require.Equal(t, derive.L1InfoEcotoneLen, len(transactions[0].Data()))

?